compiler/tests/runtime.rs

```diff
@@ -23,13 +25,37 @@ macro_rules! assert_eq {
     };
 }
 
+#[rstest(
+    reason,
+    case("foo"),
+    case("A very looooooooooooooong reason that consumes multiple words")
+)]
+fn test_revert_with_reason_string(reason: &str) {
+    let reason_id = format!(r#""{}""#, keccak::full(reason.as_bytes()));
+
+    with_executor(&|mut executor| {
+        test_runtime_functions_revert(
+            &mut executor,
+            Runtime::default()
+                .with_data(
```

Comment:
I introduced the `Runtime` object with the builder pattern to not introduce a `data` parameter to every test that most of them would not end up using.

We could actually go a bit further and incorporate the testing functions on the `Runtime` object, too so that the code becomes something like:

```
test_runtime_functions_revert(
            Runtime::default()
                .with_data(
                    vec![yul::Data { name: keccak::full(reason.as_bytes()), value: reason.to_owned() }]
                )
                .with_test_statements(
                statements! {
                    (let reason := load_data_string((dataoffset([literal_expression! { (reason_id) }])), (datasize([literal_expression! { (reason_id) }]))))
                    (revert_with_reason_string(reason))
                })
                .execute(executor)
                .expect_revert_with(reason)
        );
```

I'm happy to create an issue / PR if you happen to like it @g-r-a-n-t

Reply:
Seems like a more rusty way of doing things :+1: I'm open to refactoring this way.